Make invalid JSON examples (maybe) less confusing (#30948)
mdn/content: files/en-us/web/javascript/reference/global_objects/json/parse/index.md

- "Guess" based on the structure of the data (for example, an array of two-member arrays)
- If the shape of the payload is fixed, based on the property name (for example, all properties called `registry` hold `Map` objects).

### Illegal JSON

When `JSON.parse` receives a string that does not conform to the JSON grammar, it throws a `SyntaxError`.

Arrays and objects cannot have [trailing commas](/en-US/docs/Web/JavaScript/Reference/Trailing_commas) in JSON:

```js example-bad

> JSON.parse("[1, 2, 3, 4, ]"); // SyntaxError: Unexpected token ] in JSON at position 13

Reviewer: It might be good to put the comments on a separate line to the code lines, and then separate each pair with a blank line so it is obvious which comment is related to which code line.

That way it is generally more readable due to shorter line lengths, and you've got less chance of the lines wrapping untidily.

JSON.parse('{"foo": 1, }'); // SyntaxError: Unexpected token } in JSON at position 12
```

> String literals in JSON must be delimited by double (not single) quotes:

Reviewer: ```suggestion
Strings contained inside JSON must be delimited by double (not single) quotes:
```

```js example-bad
JSON.parse("{'foo': 1}"); // SyntaxError: Unexpected token ' in JSON at position 1
JSON.parse("'string'"); // SyntaxError: Unexpected token ' in JSON at position 0
```

> So, if you are writing JSON in a string literal, you should either use single quotes as the string literal's delimiters, or otherwise escape the double quotes in the JSON string:

Reviewer: ```suggestion
If you are writing JSON as a string literal, you should either use single quotes to delimit the entire string literal or escape the double quotes inside the JSON string:
```
